Allow listing a user's and org's campaigns

mrnugget opened the pull request:
This fixes #12563.

@eseliger does it make sense to make the `Namespace: graphql.ID` parameter on `campaigns()` public? It would allow filtering campaigns by namespace on the main page. But I'm not sure whether we have a need for that right now.

@efritz @unknwon I saw that you two use this pattern of using `EnterpriseResolvers` to talk to them from the normal graphqlbackend — can you both take a look and see whether I did it correctly?

eseliger commented on enterprise/internal/campaigns/resolvers/resolver.go:
```diff
@@ -383,6 +383,7 @@ func (r *Resolver) Campaigns(ctx context.Context, args *graphqlbackend.ListCampa
 		return nil, err
```
Just a sidenote: Line 381 comment seems outdated(?)

mrnugget replied:
The admin check? Yeah, I left that in until we tackle the issue completely. I still have another ticket (https://github.com/sourcegraph/sourcegraph/issues/10713) and PR (#11621) for that.

eseliger commented on enterprise/internal/campaigns/resolvers/resolver.go:
```diff
@@ -402,6 +404,17 @@ func (r *Resolver) Campaigns(ctx context.Context, args *graphqlbackend.ListCampa
+	if args.Namespace != nil {
+		switch relay.UnmarshalKind(*args.Namespace) {
+		case "User":
+			err = relay.UnmarshalSpec(*args.Namespace, &opts.NamespaceUserID)
```
these errors are unhandled, no?

mrnugget replied:
Fixed!

eseliger commented:
> @eseliger does it make sense to make the Namespace: graphql.ID parameter on campaigns() public? It would allow filtering campaigns by namespace on the main page. But I'm not sure whether we have a need for that right now.

I don't think we need it (now).

mrnugget commented:
I added tests, the missing error-check that @eseliger pointed out, found and fixed a bug *and* thanks to the tests I realized that `TotalCount` was wrong.

Merging this now and if there's any post-merge feedback: feel free!

eseliger commented on enterprise/internal/campaigns/resolvers/resolver.go:
```diff
@@ -402,6 +404,23 @@ func (r *Resolver) Campaigns(ctx context.Context, args *graphqlbackend.ListCampa
+			return nil, err
+		}
+	}
+
+	fmt.Printf("opts=%+v\n", opts)
+
```
leftover :)

mrnugget replied:
🤦
